src/interpreter.c

```diff
@@ -217,7 +219,9 @@ static jl_value_t *eval(jl_value_t *e, jl_value_t **locals, size_t nl)
         size_t i;
         for (i=0; i < nl; i++) {
             if (locals[i*2] == sym) {
-                return (locals[i*2+1] = eval(args[1], locals, nl));
+                locals[i*2+1] = eval(args[1], locals, nl);
+                gc_wb(jl_current_module, locals[i*2+1]); // not sure about jl_current_module
```

Comment:
since `locals` is a stack location, this `gc_wb` seems unnecessary? (or perhaps, should target `jl_current_task`?)